src/components/map.tsx

```diff
@@ -288,6 +292,70 @@ const useGeoData = (year: string) => {
   return geoData;
 };
 
+/**
+ * Get the map as an image, using the deckgl canvas and html2canvas to get
+ * the legend as an image.
+ */
+const getImageData = async (deck: Deck, legend: HTMLElement) => {
+  if (!deck || "canvas" in deck === false) {
+    return;
+  }
+
+  // @ts-expect-error canvas is private
+  const canvas = deck.canvas;
+  if (!canvas) {
+    return;
+  }
+
+  const initialSize = {
+    width: canvas.width,
+    height: canvas.height,
+  };
+
+  const wantedSize = {
+    width: 1440 * 2,
+    height: 1024 * 2,
+  };
+
+  Object.assign(canvas, wantedSize);
+  deck.redraw("New size");
+
+  const newCanvas = document.createElement("canvas");
+  const padding = 0;
+  newCanvas.width = canvas.width + padding * 2;
+  newCanvas.height = canvas.height + padding * 2;
+  const context = newCanvas.getContext("2d");
+  if (!context) {
+    return;
+  }
+
+  // Using html2canvas, take the legend element, and draw it on the new canvas
+  // Make a new canvas element to convert the image to a png
+  // We need a new canvas since we will draw the legend onto it
+  context.fillStyle = "white";
+  context.fillRect(0, 0, newCanvas.width, newCanvas.height);
+
+  context.drawImage(canvas, 0, 0, canvas.width, canvas.height);
+
+  const legendCanvas = await html2canvas(legend);
+
+  // We need to draw the legend using the device pixel ratio otherwise we get
+  // difference between different browsers (Safari legend would be bigger somehow)
+  const { width, height } = legend.getBoundingClientRect();
+  const ratio = window.devicePixelRatio;
+  context.drawImage(legendCanvas, 12, 12, width * ratio, height * ratio);
+
+  // Returns the canvas as a png
+  const res = newCanvas
+    .toDataURL("image/png")
+    .replace("image/png", "image/octet-stream");
```

Comment:
Why octet-stream? 🤔

Reply:
Changed that to use the more modern canvas.toBlob + URL.createObjectURL.

See https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL

> Warning: toDataURL() encodes the whole image in an in-memory string. For larger images, this can have performance implications, and may even overflow browsers' URL length limit when assigned to [HTMLImageElement.src](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/src). You should generally prefer [toBlob()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob) instead, in combination with [URL.createObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static).